src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -1895,11 +1970,10 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
     on_connect_follow_mode <- any(on_output_connect_follow_mode,on_input_connect_follow_mode);
     connect_drag_mode      <- any(on_connect_drag_mode,on_connect_follow_mode);
 
-    on_detached_edge    <- any(&inputs.some_edge_targets_detached,&inputs.some_edge_sources_detached);
-    has_detached_edge   <- bool(&inputs.all_edges_attached,&on_detached_edge);
-
     eval node_input_touch.down ((target)   model.frp.press_node_input.emit(target));
     eval node_output_touch.down ((target)  model.frp.press_node_output.emit(target));
+    eval node_input_touch.up ((target)   model.frp.release_node_input.emit(target));
+    eval node_output_touch.up ((target)  model.frp.release_node_output.emit(target));
```

Comment:
you should never ever emit INPUTS to FRP network. You are allowed to emit outputs only from inside of the network,. Inputs are alwyas considered public API and they should never be emitted not by external thing. Otherwise we will have logical cycles everywhere and this will become spaghetti